src/util/file.rs

```diff
@@ -97,10 +98,20 @@ mod test {
             let _ = OpenOptions::new().write(true).create_new(true).open(&existent_file).unwrap();
         }
         assert_eq!(file_exists(&existent_file), true);
-        assert_eq!(delete_file_if_exist(&existent_file), true);
+        delete_file_if_exist(&existent_file);
         assert_eq!(file_exists(&existent_file), false);
 
+        let perm_file = dir_path.join("perm_file");
```

Comment:
No, it is supported to work to delete a file with 400 perm bits. The case you propose is a positive case but not the opposite.